templates/scripts/e2e_tests/yaml_linting.sh

```diff
@@ -17,10 +17,15 @@ LINT_OUTPUT=$(jq --null-input --arg yaml "$(<${TMPFILE})" '.content=$yaml' |
     --header 'Content-Type: application/json' \
     --header "PRIVATE-TOKEN: ${LINT_CI_TOKEN}" \
     --data @-)
-echo Linting output: "${LINT_OUTPUT}"
-LINT_VAL=$(echo "$LINT_OUTPUT" | jq --raw-output '.valid')
+echo "Linting output: ${LINT_OUTPUT}"
+LINT_VAL=$(echo "${LINT_OUTPUT}" | jq --raw-output '.valid')
 if [ "${LINT_VAL}" != "true" ]; then
   echo "The linting of the YAML pipeline file failed!"
-  echo "Check the corresponding GitLab CI file and the 'Linting output' in this log."
+  if [ "$(echo "${LINT_OUTPUT}" | jq --raw-output '.errors | length')" != "0" ]; then
+    echo "The following errors were found:"
+    echo "${LINT_OUTPUT}" | jq --raw-output '.errors[] | "- " + .'
+  else
+    echo "Check the corresponding GitLab CI file and the 'Linting output' in this log."
+  fi
```

Comment:
ok I see that here we basically want to format better the errors, instead of having the raw json string above (the 'Linting output').

But, since there can be [also the warnings](https://docs.gitlab.com/ee/api/lint.html), and this nested if complicates logic, (albeit not much) what about we just simply change line 20 to something like `echo "Linting output: $(echo ${LINT_OUTPUT} | jq --raw-output .)"`

So, we always get a nicely formatted output, whether the rc is 0 or not (if the job is green no one cares looking at it anyway).

Reply:
Because now the JSON object contains also the sended YAML pipeline as a text string. I keept the original JSON (`LINT_OUTPUT`) only to have the entire object in the output if something changes in the future.

In any case we can print the JSON response in a pretty way using the `jq` command. What do you think?

Reply:
ah yes, I see, you are right, having the whole doc would make it impossible to read.  
So yes, let's do as you say. If it's ok with you I would suggest ss a possible optional improvement to skip the conditonal at line 24, since we are in a section where the validation failed, and always print errors + warning, something like (not tested):

```bash
    echo "The following errors were found:"
    echo "${LINT_OUTPUT}" | jq --raw-output '.errors[] | "- " + .'
    echo "The following warnings were found:"
    echo "${LINT_OUTPUT}" | jq --raw-output '.warnings[] | "- " + .'
    echo "Check the corresponding GitLab CI file and the 'Linting output' in this log."
```

Reply:
Changed! Thanks a lot for the suggestion!